src/dump.cpp

```diff
@@ -213,26 +209,38 @@ void game::dump_stats( const std::string& what, dump_mode mode )
             header.push_back( std::to_string( i ) );
         }
 
-        auto dump = [&rows]( const std::string &name, const standard_npc &who, const item &gun) {
-            std::vector<std::string> r( 1, name );
+        auto dump = [&rows]( const standard_npc &who, const item &gun) {
+            std::vector<std::string> r( 1, string_format( "%s %s", who.get_name().c_str(), gun.tname().c_str() ) );
             for( int i = 0; i <= cycles; ++i ) {
                 r.push_back( string_format( "%.2f", who.gun_engagement_range( gun, i ) ) );
             }
             rows.push_back( r );
         };
 
-        item g19( "glock_19" );
-        dump( "Glock", s1, g19.ammo_set( "9mm" ) );
-
-        item mp5( "hk_mp5" );
-        dump( "MP5", s1, mp5.ammo_set( "9mm" ) );
+        if( opts.empty() ) {
+            dump( test_npcs[ "S1" ], test_items[ "G1" ] );
+            dump( test_npcs[ "S1" ], test_items[ "G2" ] );
+            dump( test_npcs[ "S1" ], test_items[ "G3" ] );
+            dump( test_npcs[ "S1" ], test_items[ "G4" ] );
+
+        } else {
+            for( const auto &str : opts ) {
+                auto idx = str.find( ':' );
+                if( idx == std::string::npos ) {
+                    std::cerr << "cannot parse test case: " << str << std::endl;
+                    return false;
+                }
+                auto s = test_npcs.find( str.substr( 0, idx ) );
+                auto g = test_items.find( str.substr( idx + 1 ) );
```

Comment:
Since this name is used by our global variable, it would be better to rename this one. Though it has local scope, you cant use the global here (game.h is included so it's possible). Also, all occurrences of `g->whatever` look really misleading. I'd consider `g` as a reserved name.